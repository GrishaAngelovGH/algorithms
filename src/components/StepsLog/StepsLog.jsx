const StepsLog = ({ title, children, onClose }) => (
	<div className='offcanvas offcanvas-end show' tabIndex='-1' id='offcanvas' aria-labelledby='offcanvasLabel' >
		<div className='offcanvas-header bg-primary text-white'>
			<h5 className='offcanvas-title' id='offcanvasLabel'>{title}</h5>
			<button type='button' className='btn-close bg-light' data-bs-dismiss='offcanvas' aria-label='Close' onClick={onClose}></button>
		</div>
		<div className='offcanvas-body'>
			{children}
		</div>
	</div>
)

export default StepsLog