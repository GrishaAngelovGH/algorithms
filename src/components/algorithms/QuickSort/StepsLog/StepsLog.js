const StepsLog = ({ onClose }) => (
    <div className="offcanvas offcanvas-end show" tabIndex="-1" id="offcanvas" aria-labelledby="offcanvasLabel" >
        <div className="offcanvas-header">
            <h5 className="offcanvas-title" id="offcanvasLabel">Quick Sort Steps</h5>
            <button type="button" className="btn-close" data-bs-dismiss="offcanvas" aria-label="Close" onClick={onClose}></button>
        </div>
        <div className="offcanvas-body">
            <p className='fw-bold'>[ 5, 1, 3, 2, 4 ]</p>
            <p className='fw-bold'>Pivot: 3</p>
            <p className='fw-bold'>Swap 3 &#128257; 4</p>
            <p className='fw-bold'>[ 5, 1, 4, 2, 3 ]</p>
            <p className='fw-bold'>Partitioning for [5, 2]</p>
            <p className='fw-bold'>Partitioning set left pointer: 5</p>
            <p className='fw-bold'>Partitioning set right pointer: 2</p>
            <p className='fw-bold'>Partitioning will swap 5 &#128257; 2</p>
            <p className='fw-bold'>Partitioning set left pointer: 4</p>
            <p className='fw-bold'>Partitioning set right pointer: 1</p>
            <p className='fw-bold'>Partitioning marks 4</p>
            <p className='fw-bold'>Swap 4 &#128257; 3</p>
            <p className='fw-bold'>[ 2, 1, 3, 5, 4 ]</p>
            <p className='fw-bold'>{`Index check: 2-0 > 1 (true)`}</p>
            <p className='fw-bold'>Sort left partition [2, 1]</p>
            <hr />
            <p className='fw-bold'>[ 2, 1, 3, 5, 4 ]</p>
            <p className='fw-bold'>Pivot: 2</p>
            <p className='fw-bold'>Swap 2 &#128257; 1</p>
            <p className='fw-bold'>[ 1, 2, 3, 5, 4 ]</p>
            <p className='fw-bold'>Partitioning for [1, 1]</p>
            <p className='fw-bold'>Partitioning set left pointer: 2</p>
            <p className='fw-bold'>Partitioning set right pointer: 1</p>
            <p className='fw-bold'>Partitioning marks 2</p>
            <p className='fw-bold'>Swap 2 &#128257; 2</p>
            <p className='fw-bold'>[ 1, 2, 3, 5, 4 ]</p>
            <p className='fw-bold'>{`Index check: 4-2 > 1 (true)`}</p>
            <p className='fw-bold'>Sort right partition [5, 4]</p>
            <hr />
            <p className='fw-bold'>[ 1, 2, 3, 5, 4 ]</p>
            <p className='fw-bold'>Pivot: 5</p>
            <p className='fw-bold'>Swap 5 &#128257; 4</p>
            <p className='fw-bold'>[ 1, 2, 3, 4, 5 ]</p>
            <p className='fw-bold'>Partitioning for [4, 4]</p>
            <p className='fw-bold'>Partitioning set left pointer: 5</p>
            <p className='fw-bold'>Partitioning set right pointer: 4</p>
            <p className='fw-bold'>Partitioning marks 5</p>
            <p className='fw-bold'>Swap 5 &#128257; 5</p>
            <p className='fw-bold'>[ 1, 2, 3, 4, 5 ]</p>
            <p className='fw-bold'>Result: [ 1, 2, 3, 4, 5 ]</p>        </div>
    </div>
)

export default StepsLog