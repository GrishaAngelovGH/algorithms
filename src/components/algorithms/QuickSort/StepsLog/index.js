import StepsLog from './StepsLog'

export default StepsLog