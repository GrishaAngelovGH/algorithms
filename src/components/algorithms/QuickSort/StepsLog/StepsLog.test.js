import { render } from '@testing-library/react'

import StepsLog from './StepsLog'

test('should render component', () => {
    const container = render(<StepsLog />)

    expect(container).toMatchSnapshot()
})