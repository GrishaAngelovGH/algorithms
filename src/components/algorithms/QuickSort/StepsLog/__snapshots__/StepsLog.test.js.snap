// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-labelledby="offcanvasLabel"
        class="offcanvas offcanvas-end show"
        id="offcanvas"
        tabindex="-1"
      >
        <div
          class="offcanvas-header"
        >
          <h5
            class="offcanvas-title"
            id="offcanvasLabel"
          >
            Quick Sort Steps
          </h5>
          <button
            aria-label="Close"
            class="btn-close"
            data-bs-dismiss="offcanvas"
            type="button"
          />
        </div>
        <div
          class="offcanvas-body"
        >
          <p
            class="fw-bold"
          >
            [ 5, 1, 3, 2, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Pivot: 3
          </p>
          <p
            class="fw-bold"
          >
            Swap 3 游대 4
          </p>
          <p
            class="fw-bold"
          >
            [ 5, 1, 4, 2, 3 ]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning for [5, 2]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set left pointer: 5
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set right pointer: 2
          </p>
          <p
            class="fw-bold"
          >
            Partitioning will swap 5 游대 2
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set left pointer: 4
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set right pointer: 1
          </p>
          <p
            class="fw-bold"
          >
            Partitioning marks 4
          </p>
          <p
            class="fw-bold"
          >
            Swap 4 游대 3
          </p>
          <p
            class="fw-bold"
          >
            [ 2, 1, 3, 5, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Index check: 2-0 &gt; 1 (true)
          </p>
          <p
            class="fw-bold"
          >
            Sort left partition [2, 1]
          </p>
          <hr />
          <p
            class="fw-bold"
          >
            [ 2, 1, 3, 5, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Pivot: 2
          </p>
          <p
            class="fw-bold"
          >
            Swap 2 游대 1
          </p>
          <p
            class="fw-bold"
          >
            [ 1, 2, 3, 5, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning for [1, 1]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set left pointer: 2
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set right pointer: 1
          </p>
          <p
            class="fw-bold"
          >
            Partitioning marks 2
          </p>
          <p
            class="fw-bold"
          >
            Swap 2 游대 2
          </p>
          <p
            class="fw-bold"
          >
            [ 1, 2, 3, 5, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Index check: 4-2 &gt; 1 (true)
          </p>
          <p
            class="fw-bold"
          >
            Sort right partition [5, 4]
          </p>
          <hr />
          <p
            class="fw-bold"
          >
            [ 1, 2, 3, 5, 4 ]
          </p>
          <p
            class="fw-bold"
          >
            Pivot: 5
          </p>
          <p
            class="fw-bold"
          >
            Swap 5 游대 4
          </p>
          <p
            class="fw-bold"
          >
            [ 1, 2, 3, 4, 5 ]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning for [4, 4]
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set left pointer: 5
          </p>
          <p
            class="fw-bold"
          >
            Partitioning set right pointer: 4
          </p>
          <p
            class="fw-bold"
          >
            Partitioning marks 5
          </p>
          <p
            class="fw-bold"
          >
            Swap 5 游대 5
          </p>
          <p
            class="fw-bold"
          >
            [ 1, 2, 3, 4, 5 ]
          </p>
          <p
            class="fw-bold"
          >
            Result: [ 1, 2, 3, 4, 5 ]
          </p>
                  
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-labelledby="offcanvasLabel"
      class="offcanvas offcanvas-end show"
      id="offcanvas"
      tabindex="-1"
    >
      <div
        class="offcanvas-header"
      >
        <h5
          class="offcanvas-title"
          id="offcanvasLabel"
        >
          Quick Sort Steps
        </h5>
        <button
          aria-label="Close"
          class="btn-close"
          data-bs-dismiss="offcanvas"
          type="button"
        />
      </div>
      <div
        class="offcanvas-body"
      >
        <p
          class="fw-bold"
        >
          [ 5, 1, 3, 2, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Pivot: 3
        </p>
        <p
          class="fw-bold"
        >
          Swap 3 游대 4
        </p>
        <p
          class="fw-bold"
        >
          [ 5, 1, 4, 2, 3 ]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning for [5, 2]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set left pointer: 5
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set right pointer: 2
        </p>
        <p
          class="fw-bold"
        >
          Partitioning will swap 5 游대 2
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set left pointer: 4
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set right pointer: 1
        </p>
        <p
          class="fw-bold"
        >
          Partitioning marks 4
        </p>
        <p
          class="fw-bold"
        >
          Swap 4 游대 3
        </p>
        <p
          class="fw-bold"
        >
          [ 2, 1, 3, 5, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Index check: 2-0 &gt; 1 (true)
        </p>
        <p
          class="fw-bold"
        >
          Sort left partition [2, 1]
        </p>
        <hr />
        <p
          class="fw-bold"
        >
          [ 2, 1, 3, 5, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Pivot: 2
        </p>
        <p
          class="fw-bold"
        >
          Swap 2 游대 1
        </p>
        <p
          class="fw-bold"
        >
          [ 1, 2, 3, 5, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning for [1, 1]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set left pointer: 2
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set right pointer: 1
        </p>
        <p
          class="fw-bold"
        >
          Partitioning marks 2
        </p>
        <p
          class="fw-bold"
        >
          Swap 2 游대 2
        </p>
        <p
          class="fw-bold"
        >
          [ 1, 2, 3, 5, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Index check: 4-2 &gt; 1 (true)
        </p>
        <p
          class="fw-bold"
        >
          Sort right partition [5, 4]
        </p>
        <hr />
        <p
          class="fw-bold"
        >
          [ 1, 2, 3, 5, 4 ]
        </p>
        <p
          class="fw-bold"
        >
          Pivot: 5
        </p>
        <p
          class="fw-bold"
        >
          Swap 5 游대 4
        </p>
        <p
          class="fw-bold"
        >
          [ 1, 2, 3, 4, 5 ]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning for [4, 4]
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set left pointer: 5
        </p>
        <p
          class="fw-bold"
        >
          Partitioning set right pointer: 4
        </p>
        <p
          class="fw-bold"
        >
          Partitioning marks 5
        </p>
        <p
          class="fw-bold"
        >
          Swap 5 游대 5
        </p>
        <p
          class="fw-bold"
        >
          [ 1, 2, 3, 4, 5 ]
        </p>
        <p
          class="fw-bold"
        >
          Result: [ 1, 2, 3, 4, 5 ]
        </p>
                
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
